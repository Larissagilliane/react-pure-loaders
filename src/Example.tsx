import * as React from 'react';

export const Example = (props: any) => {
  return <span {...props}>Hello World!</span>;
};