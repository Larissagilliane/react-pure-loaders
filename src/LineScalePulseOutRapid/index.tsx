import { css } from 'glamor';
import * as React from 'react';
import style from './style';

export interface IGeneralPropTypes {
  loading: boolean;
}

export class LineScalePulseOutRapid extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { loading } = this.props;

    if (loading) {
      return (
        <div {...css(style)}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
