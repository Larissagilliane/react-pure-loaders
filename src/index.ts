/**
 * Dots
 */
export * from './BallBeat';
export * from './BallClipRotate';
export * from './BallClipRotateMultiple';
export * from './BallClipRotatePulse';
export * from './BallGridBeat';
export * from './BallGridPulse';
export * from './BallPulse';
export * from './BallPulseRise';
export * from './BallPulseRound';
export * from './BallPulseSync';
export * from './BallRotate';
export * from './BallScale';
export * from './BallScaleMultiple';
export * from './BallScaleRandom';
export * from './BallScaleRipple';
export * from './BallScaleRippleMultiple';
export * from './BallSpinFadeLoader';
export * from './BallSpinLoader';
export * from './BallTrianglePath';
export * from './BallZigZag';
export * from './BallZigZagDeflect';

/**
 * Lines
 */
export * from './LineScale';
export * from './LineScaleParty';
export * from './LineScalePulseOut';
export * from './LineScalePulseOutRapid';
export * from './LineScaleRandom';
export * from './LineSpinFadeLoader';

/**
 * Misc
 */
export * from './CubeTransition';
export * from './Pacman';
export * from './SemiCircleSpin';
export * from './SquareSpin';
export * from './TriangleSkewSpin';
