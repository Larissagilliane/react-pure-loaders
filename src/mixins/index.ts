import { css } from 'glamor';
import {
  BALL_SIZE,
  LINE_HEIGHT,
  LINE_WIDTH,
  MARGIN,
  PRIMARY_COLOR,
} from '../variables';

interface IGrid {
  [id: string]: object;
}

export function globalBg() {
  return {
    backgroundColor: PRIMARY_COLOR,
  };
}

export function globalAnimation() {
  return {
    animationFillMode: 'both',
  };
}

export function balls() {
  return css(
    {
      borderRadius: '100%',
      height: BALL_SIZE,
      margin: MARGIN,
      width: BALL_SIZE,
    },
    globalBg(),
  );
}

export function ballGrid() {
  let i = 1;
  const grid: IGrid = {};
  for (; i < 10; i += 1) {
    const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
    const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.6);

    grid[` > div:nth-child(${i})`] = {
      animationDelay: `${delay}s`,
      animationDuration: `${duration}s`,
    };
  }

  return grid;
}

export function lines() {
  return css(
    {
      borderRadius: '2px',
      height: LINE_HEIGHT,
      margin: MARGIN,
      width: LINE_WIDTH,
    },
    globalBg(),
  );
}
