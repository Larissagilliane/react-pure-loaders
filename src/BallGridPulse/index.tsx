import { css } from 'glamor';
import * as React from 'react';
import { IGeneralPropTypes } from '../mixins';
import style from './style';

export class BallGridPulse extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { color, loading } = this.props;

    if (loading) {
      return (
        <div {...css(style(color))}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
