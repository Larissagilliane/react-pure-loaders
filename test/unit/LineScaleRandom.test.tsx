import { mount, render, shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { StyleSheet } from 'glamor/lib/sheet';
import * as React from 'react';
import { LineScaleRandom } from '../../src';

const stylesheet = new StyleSheet();

describe('<LineScaleRandom>', () => {
  beforeEach(() => {
    stylesheet.flush();
  });

  test('<LineScaleRandom> Shallow', () => {
    // const wrapper = render(<LineScaleRandom loading={true}/>);
    //
    // expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
  test('<LineScaleRandom> Mount', () => {
    // const ui = (<LineScaleRandom loading={true}/>);
    //
    // expect(enzymeToJson(mount(ui))).toMatchSnapshot();
  });
  test('<LineScaleRandom> Render', () => {
    // const ui = (<LineScaleRandom loading={true}/>);
    //
    // expect(enzymeToJson(render(ui))).toMatchSnapshot();
  });
});
