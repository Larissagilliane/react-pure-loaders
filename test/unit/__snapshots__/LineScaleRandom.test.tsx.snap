// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScaleRandom> LineScaleRandom should match snapshot 1`] = `
.emotion-0 > div:nth-of-type(1) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
}

.emotion-0 > div:nth-of-type(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
}

.emotion-0 > div:nth-of-type(3) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
}

.emotion-0 > div:nth-of-type(4) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
}

.emotion-0 > div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
